lib/game.rb

```diff
@@ -0,0 +1,170 @@
+          user_input = gets.chomp.upcase.split(", ")
+      until @player_board.valid_placement?(ship, user_input)
+          puts "Invalid placement. Must be consecutive and not diagonal"
+          user_input = gets.chomp.upcase.split(", ")
+      end
+      @player_board.place_ship(ship, user_input)
+      puts " \n Your #{ship.name} has been placed. \n "
+  end
+
+  def turn_start
+      puts "==========COMPUTER BOARD========= \n "
+      puts @computer_board.render
+      puts " \n ==========PLAYER BOARD=========== \n "
+      puts @player_board.render(true)
+  end
+  
+  def player_turn_shot
+      puts " \n Enter the coordinate for your shot: \n "
+          user_input = gets.chomp.upcase
+      until @computer_board.valid_coordinate?(user_input) && !@computer_board.cells[user_input].fired_upon?
+          puts "You have already fired here.  Enter a new coordinate:"
+          user_input = gets.chomp.upcase
+      end
+      #this could be put into a helper method and then on line 82.5 we call it
+          @computer_board.cells[user_input].fire_upon
+      puts @computer_board.render
+      outcome = @computer_board.cells[user_input].render
+      shot_message = case outcome
+      when "M" then "was a miss."
+      when "H" then "scored a hit!"
+      when "X" then "sunk my battleship!"
+      else "Oh no, something has gone wrong"
+      end
+      puts "Your shot on #{user_input} #{shot_message} \n "
+  end
+
+  def computer_turn_shot
+          random_coords = @player_board.cells.keys.sample
+      until @player_board.valid_coordinate?(random_coords) && !@player_board.cells[random_coords].fired_upon?
+          random_coords = @player_board.cells.keys.sample
+      end
+          @player_board.cells[random_coords].fire_upon
+      puts @player_board.render(true)
+      computer_outcome = @player_board.cells[random_coords].render
+      computer_shot_message = case computer_outcome
+          when "M" then "was a miss."
+          when "H" then "scored a hit!"
+          when "X" then "sunk your battleship!"
+          else "Oh no, something has gone wrong"
+          end 
+      puts "My shot on #{random_coords} #{computer_shot_message} \n "
+  end
+
+  def play_game
+      until game_over?
+          turn_start
+          player_turn_shot
+      break if player_won?
+          computer_turn_shot
+      end
+      end_game
+  end
+
+  def game_over?
+    player_won? || computer_won?
+  end
+
+  def player_won?
+    [@computer_cruiser, @computer_submarine].all?(&:sunk?)
+  end
+   
+  def computer_won?
+    [@player_cruiser, @player_submarine].all?(&:sunk?)
+  end
+
+
+  def end_game
+    if computer_won?
+         puts "I Won! #{@player_name}, would you like to play again? (yes or no) \n "
+    else
+        puts "#{@player_name}, You Won! Would you like to play again? (yes or no) \n "
+    end
+    choice = gets.chomp.downcase
+    case choice
+    when "yes",
```

Comment:
I noticed this comma when looking over this method. It may have something to do with the bug you mentioned, I will pull down merge the PR and play with the syntax here, will report back in our stand up today (7/22)